Mars Onboarding/Pages/LanguagesPage.cs

```diff
@@ -0,0 +1,137 @@
+﻿using NUnit.Framework;
+using OpenQA.Selenium.Support.UI;
+using OpenQA.Selenium;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Mars_Onboarding.Pages
+{
+    public class LanguagesPage
+    {
+
+        public void AddLanguage(IWebDriver driver)
+        {
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(20));
+            IWebElement addLanguageButton = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/thead/tr/th[3]/div")));
+            addLanguageButton.Click();
+            IWebElement languageField = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.Name("name")));
+            languageField.SendKeys("abc");
+            IWebElement levelDropdown = driver.FindElement(By.Name("level"));
+            levelDropdown.Click();
+            SelectElement selectElement = new SelectElement(levelDropdown);
+            selectElement.SelectByText("Basic");
+            IWebElement addButton = driver.FindElement(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/div/div[3]/input[1]"));
+            addButton.Click();
+
+        }
+
+        public void GetLanguage(IWebDriver driver)
+        {
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(20));
+            IWebElement languageElement = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[last()]/tr/td[1]")));
+            Assert.That(languageElement.Text == "abc", Is.True, "Test Failed: Language not found.");
+        }
+
+        public void EditLanguage(IWebDriver driver)
+        {
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(20));
+            IWebElement editLanguageButton = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementToBeClickable(By.XPath("//*[@id=\"account-profile-section\"]/div/section[2]/div/div/div/div[3]/form/div[2]/div/div[2]/div/table/tbody[last()]/tr/td[3]/span[1]/i")));
+            editLanguageButton.Click();
+            IWebElement languageField = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.Name("name")));
+            languageField.Clear();
+            languageField.SendKeys("xyz");
```

Comment:
you should first select language you want to update and then update this! also the values should be coming from the feature file not from within the page